extensions/charts/config.go

```diff
@@ -0,0 +1,9 @@
+package charts
+
+const (
+	ConfigurationFileKey = "chartUpgrade"
+)
+
+type Config struct {
+	IsUpgradable bool `json:"isUpgradable" yaml:"isUpgradable"`
```

Comment:
I need more explanation on why this is added now. It's not quite clear.

Reply:
By default when installing a chart, we install the latest version. We need a way to know when the user wants to install a version of the chart that can be upgraded i.e. by installing latest-1 version so that the chart can be upgraded later.

Reply:
I've got the same confusion, we have upgrade test cases for other chart tests, why do we need to run the same test with a different test case,  is there any difference with the existing chart test cases?

Reply:
The big difference here is cloud provider is fundamentally closer to kubernetes_version than it is to a normal chart install / upgrade. So for provisioning, the upgrade packages depends on a cluster being provided. Therefore the cloud provider depends on a cluster being provided that has charts ready for upgrade. This introduced the reason for the new variable to be used during provisioning, because we need a way to install an upgradable (latest - 1 version) chart during provisioning so that we can run an upgrade test later. 

I met with Israel and he understood this, but if its still unclear / pushback I can sync with you as well when you have time.